vpr/src/draw/search_bar.cpp

```diff
@@ -0,0 +1,287 @@
+            int ipin = device_ctx.rr_nodes[rr_node_id].ptc_num();
+            float xcen, ycen;
+
+            int iside;
+            for (iside = 0; iside < 4; iside++) {
+                if (type->pinloc[width_offset][height_offset][iside][ipin]) {
+                    draw_get_rr_pin_coords(rr_node_id, &xcen, &ycen);
+                        rr_node = {{xcen - draw_coords->pin_size, ycen - draw_coords->pin_size},
+                                {xcen + draw_coords->pin_size, ycen + draw_coords->pin_size}};
+                }
+            }
+            break;
+        }
+        case CHANX:
+        case CHANY:
+        {
+                rr_node = draw_get_rr_chan_bbox(rr_node_id);
+            break;
+        }
+        default:
+            break;
+    }
+    
+    // zoom to the node
+    ezgl::point2d offset = {rr_node.width()*1.5, rr_node.height()*1.5};
+    ezgl::rectangle zoom_view = {rr_node.m_first - offset, rr_node.m_second + offset};
+    (application.get_canvas(application.get_main_canvas_id()))->get_camera().set_world(zoom_view);
+    
+}
+
+void highlight_blocks(ClusterBlockId clb_index) {    
+    char msg[vtr::bufsize];
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& place_ctx = g_vpr_ctx.placement();
+    
+    /// determine block ///
+    ezgl::rectangle clb_bbox;
+
+    VTR_ASSERT(clb_index != EMPTY_BLOCK_ID);
+    
+    ezgl::point2d point_in_clb = clb_bbox.bottom_left();
+    highlight_sub_block(point_in_clb, clb_index, cluster_ctx.clb_nlist.block_pb(clb_index));
+    
+    
+    if (get_selected_sub_block_info().has_selection()) {
+        t_pb* selected_subblock = get_selected_sub_block_info().get_selected_pb();
+        sprintf(msg, "sub-block %s (a \"%s\") selected",
+                selected_subblock->name, selected_subblock->pb_graph_node->pb_type->name);
+    } else {
+        /* Highlight block and fan-in/fan-outs. */
+        draw_highlight_blocks_color(cluster_ctx.clb_nlist.block_type(clb_index), clb_index);
+        sprintf(msg, "Block #%zu (%s) at (%d, %d) selected.", size_t(clb_index), cluster_ctx.clb_nlist.block_name(clb_index).c_str(), place_ctx.block_locs[clb_index].loc.x, place_ctx.block_locs[clb_index].loc.y);
+    }
+    
+    application.update_message(msg);
+    
+    application.refresh_drawing();
+}
+
+
+void highlight_nets(ClusterNetId net_id) {
+    t_trace *tptr;
+    auto& route_ctx = g_vpr_ctx.routing();
+    
+    t_draw_state* draw_state = get_draw_state_vars();
+    
+    if(route_ctx.trace.size() == 0) return;
+    
+    for (tptr = route_ctx.trace[net_id].head; tptr != nullptr; tptr = tptr->next) {
+        draw_state->net_color[net_id] = ezgl::MAGENTA;
+    }
+}
+
+
+void highlight_nets(std::string net_name) {
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    ClusterNetId net_id = ClusterNetId::INVALID();
+    
+    for (auto i : cluster_ctx.clb_nlist.nets()) {
+        if(cluster_ctx.clb_nlist.net_name(i).c_str() == net_name){
```

Comment:
I think this loop can be done directly with:
```
net_id = cluster_ctx.clb_nlist.find_net(net_name)
```
Although you'll want to include a check for whether the net was found.
Likely something similar can be done with the clustered netlists `find_block(block_name)`.